
From: Nick Piggin <nickpiggin@yahoo.com.au>

I removed some stuff.  Some of them I know are fixed, some I assumed (eg. 
security), and some were insignificant items that belong on someone's
personal todo list.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/Documentation/must-fix.txt   |   39 ---------
 25-akpm/Documentation/should-fix.txt |  137 ++---------------------------------
 2 files changed, 9 insertions(+), 167 deletions(-)

diff -puN Documentation/must-fix.txt~mustfix-lists Documentation/must-fix.txt
--- 25/Documentation/must-fix.txt~mustfix-lists	2004-05-27 23:23:10.462861976 -0700
+++ 25-akpm/Documentation/must-fix.txt	2004-05-27 23:23:10.468861064 -0700
@@ -48,20 +48,6 @@ o viro: parport is nearly as bad as that
   IMO parport is more of "figure out what API changes are needed for its
   users, get them done ASAP, then fix generic layer at leisure"
 
-o (Albert Cahalan) Lots of people (check Google) get this message from the
-  kernel:
-
-  psmouse.c: Lost synchronization, throwing 2 bytes away.
-
-  (the number of bytes will be 1, 2, or 3)
-
-  At work, I get it when there is heavy NFS traffic.  The mouse goes crazy,
-  jumping around and doing random cut-and-paste all over everything.  This
-  is with a decently fast and modern PC.
-
-o There seem to be too many reports of keyboards and mice failing or acting
-  strangely.
-
 
 drivers/misc/
 ~~~~~~~~~~~~~
@@ -142,28 +128,18 @@ o Convert
 fs/
 ~~~
 
-o AIO/direct-IO writes can race with truncate and wreck filesystems.
-  (Badari has a patch)
-
 o viro: fs/char_dev.c needs removal of aeb stuff and merge of cdev-cidr.
   In progress.
 
-o forward-port sct's O_DIRECT fixes (Badari has a patch)
-
 o viro: there is some generic stuff for namei/namespace/super, but that's a
   slow-merge and can go in 2.6 just fine
 
-o trond: NFS has a mmap-versus-truncate problem (fixed? needs testing)
-
-o trond: NFSv4 client, bugs in lockd, RPSEC_GSS for NFSv[23], some atomic open
-  bits. more info: http://www.fys.uio.no/~trondmy/src/Linux-2.6.x/2.6.0-test11/
 
 kernel/sched.c
 ~~~~~~~~~~~~~~
 
 o Starvation, general interactivity need close monitoring.
 
-o SMT aware scheduler (Ingo, Rusty, Nick have implementations)
 
 kernel/
 ~~~~~~~
@@ -178,11 +154,6 @@ o Alan: 32bit uid support is *still* bro
 
 o viro: core sysctl code is racy.  And its interaction wiuth sysfs
 
-o (ingo) rwsems (on x86) are limited to 32766 waiting processes.  This
-  means that setting pid_max to above 32K is unsafe :-(
-
-  An option is to use CONFIG_RWSEM_GENERIC_SPINLOCK variant all the time,
-  for all archs, and not inline any part of the ops.
 
 lib/kobject.c
 ~~~~~~~~~~~~~
@@ -272,16 +243,6 @@ sound/
 global
 ~~~~~~
 
-o viro: 64-bit dev_t (not a mustfix for 2.6.0). 32-bit dev_t is done, 64-bit
-  means extra work on nfsd/raid/etc.
-
-o alan: Forward port 2.4 fixes
-  - Chris Wright: Security fixes including execve holes, execve vs proc races
-
-o There are about 60 or 70 security related checks that need doing
-  (copy_user etc) from Stanford tools.  (badari is looking into this, and
-  hollisb)
-
 o A couple of hundred real looking bugzilla bugs
 
 o viro: cdev rework. Mostly done.
diff -puN Documentation/should-fix.txt~mustfix-lists Documentation/should-fix.txt
--- 25/Documentation/should-fix.txt~mustfix-lists	2004-05-27 23:23:10.463861824 -0700
+++ 25-akpm/Documentation/should-fix.txt	2004-05-27 23:23:10.468861064 -0700
@@ -10,11 +10,6 @@ PRI3:	Not very important
 drivers/block/
 ~~~~~~~~~~~~~~
 
-o viro: paride drivers need a big cleanup. Partially done, but ATAPI drivers
-  need serious work and bug fixing.
-
-  PRI2
-
 drivers/char/rtc/
 ~~~~~~~~~~~~~~~~~
 
@@ -77,7 +72,7 @@ fs/
 ~~~
 
 o ext3 and ext2 block allocators have serious failure modes - interleaved
-  allocations.
+  allocations. (Reservation patch in -mm).
 
   PRI3
 
@@ -100,14 +95,6 @@ o drepper: various filesystems use ->pid
 
   PRI1
 
-o hch: devfs: there's a fundamental lookup vs devfsd race that's only
-  fixable by introducing a lookup vs devfs deadlock.  I can't see how this is
-  fixable without getting rid of the current devfsd design.  Mandrake seems
-  to have a workaround for this so this is at least not triggered so easily,
-  but that's not what I'd consider a fix..
-
-  PRI2
-
 kernel/
 ~~~~~~~
 
@@ -122,10 +109,6 @@ o rusty: Fix module-failed-init races by
 
   PRI1
 
-o Integrate userspace irq balancing daemon.
-
-  PRI2
-
 o kexec.  Seems to work, was in -mm.
 
   PRI3
@@ -137,16 +120,10 @@ o rmk: lib/inflate.c must not use static
 
   PRI2
 
-o klibc merge?
-
-  PRI2
 
 mm/
 ~~~
 
-o dropbehind for large files
-
-  PRI2
 
 net/
 ~~~~
@@ -200,7 +177,7 @@ o Pat and Pavel disagree over swsusp. Ne
 
   PRI2
 
-o Frame buffer restore codepaths (that requires some deep PCI magic)
+o Frame buffer, AGP, DRI restore.
 
   PRI2
 
@@ -208,17 +185,6 @@ o XFree86 hooks
 
   PRI2
 
-o AGP restoration
-
-  PRI2
-
-o DRI restoration
-
-  (davej/Alan: not super-critical, can crash laptop on restore.  davej
-  looking into it.)
-
-  PRI2
-
 o IDE suspend/resume without races (Ben is looking at this a little)
 
   PRI2
@@ -233,6 +199,9 @@ o Pat: There are already CPU device stru
 global
 ~~~~~~
 
+o viro: 64-bit dev_t (not a mustfix for 2.6.0). 32-bit dev_t is done, 64-bit
+  means extra work on nfsd/raid/etc.
+
 o We need a kernel side API for reporting error events to userspace (could
   be async to 2.6 itself)
 
@@ -256,9 +225,6 @@ o Alan: Cardbus/PCMCIA requires all Russ
 drivers/acpi/
 ~~~~~~~~~~~~~
 
-o Fix acpi for all newer IBM Thinkpads see
-  http://bugme.osdl.org/show_bug.cgi?id=1038 for more information
-
 o alan: VIA APIC stuff is one bit of this, there are also some other
   reports that were caused by ACPI not setting level v edge trigger some
   times
@@ -285,9 +251,6 @@ o mochel: ACPI suspend doesn't work.  Im
 drivers/block/
 ~~~~~~~~~~~~~~
 
-o More testing of floppy
-
-  PRI3
 
 drivers/char/
 ~~~~~~~~~~~~~
@@ -298,49 +261,20 @@ drivers/ide/
 
   (Alan)
 
-o IDE PIO has occasional unexplained PIO disk eating reports
-
-  PRI1
-
-o IDE has multiple zillions of races/hangs in 2.5 still
-
-  PRI1
-
-o IDE scsi needs rewriting
-
-  PRI2
-
-o IDE needs significant reworking to handle Simplex right
-
-  PRI2
-
-o IDE hotplug handling for 2.5 is completely broken still
+o IDE races, PIO problems, simplex, hotplug, taskfile.
 
   PRI2
 
-o There are lots of other IDE bugs that wont go away until the taskfile
-  stuff is included, the locking bugs that allow any user to hang the IDE
-  layer in 2.5, and some other updates are forward ported.  (esp.  HPT372N).
-
-  PRI1
 
 drivers/isdn/
 ~~~~~~~~~~~~~
 
   (Kai, rmk)
 
-o isdn_tty locking is completely broken (cli() and friends)
-
-  PRI2
-
-o fix other drivers
+o locking fixes, cleanups, adaption to recent APIs etc
 
   PRI2
 
-o lots more cleanups, adaption to recent APIs etc
-
-  PRI3
-
 o fixup tty-based ISDN drivers which provide TIOCM* ioctls (see my recent
   3-set patch for serial stuff)
 
@@ -389,14 +323,9 @@ o hch, Mike Anderson, Badari Pulavarty: 
 sound/
 ~~~~~~
 
-o rmk: several OSS drivers for SA11xx-based hardware in need of
-  ALSA-ification and L3 bus support code for these.
-
-o rmk: need to complete ALSA-ification of the WaveArtist driver for both
-  NetWinder and other stuff (there's some fairly fundamental differences in
-  the way the mixer needs to be handled for the NetWinder.)
+  (rmk)
 
-  (Issues with forward-porting 2.4 bugfixes.)
+o ALSA-fication of drivers, forward port 2.4 bugfixes
   (Killing off OSS is 2.7 material)
 
 PRI2
@@ -408,23 +337,6 @@ o davej: PAT support (for mtrr exhaustio
 
   PRI2
 
-o 2.5.x won't boot on some 440GX
-
-  alan: Problem understood now, feasible fix in 2.4/2.4-ac.  (440GX has two
-  IRQ routers, we use the $PIR table with the PIIX, but the 440GX doesnt use
-  the PIIX for its IRQ routing).  Fall back to BIOS for 440GX works and Intel
-  concurs.
-
-  PRI1
-
-o 2.5.x doesn't handle VIA APIC right yet.
-
-  1. We must write the PCI_INTERRUPT_LINE
-
-  2. We have quirk handlers that seem to trash it.
-
-  PRI1
-
 o ECC driver questions are not yet sorted (DaveJ is working on this) (Dan
   Hollis)
 
@@ -434,29 +346,12 @@ o ECC driver questions are not yet sorte
 
   PRI3
 
-o alan: 2.4 has some fixes for tsc handling bugs.  One where some bioses in
-  SMM mode mess up our toggle on the time high/low or mangle the counter and
-  one where a few chips need religious use of _p for timer access and we
-  don't do that.  This is forward porting little bits of fixup.
-
-  ACPI HZ stuff we can't trap - a lot of ACPI is implemented as outb's
-  triggering SMM traps
-
-  PRI1
 
 arch/x86_64/
 ~~~~~~~~~~~~
 
   (Andi)
 
-o time handling is broken. Need to move up 2.4 time.c code.
-
-  PRI1
-
-o NMI watchdog seems to tick too fast
-
-  PRI2
-
 o need to coredump 64bit vsyscall code with dwarf2
 
   PRI2
@@ -499,19 +394,10 @@ o SH needs resyncing, as do some other p
 arch/s390/
 ~~~~~~~~~
 
-o A nastly memory management problem causes random crashes.  These appear
-  to be fixed/hidden by the objrmap patch, more investigation is needed.
-
-  PRI1
 
 drivers/s390/
 ~~~~~~~~~~~~~
 
-o Early userspace and 64 bit dev_t will allow the removal of most of
-  dasd_devmap.c and dasd_genhd.c.
-
-  PRI2
-
 o The 3270 console driver needs to be replaced with a working one
   (prototype is there, needs to be finished).
 
@@ -522,11 +408,6 @@ o Minor interface changes are pending in
 
   PRI2
 
-o Jan Glauber is working on a fix for the timer issues related to running
-  on virtualized CPUs (wall-clock vs.  cpu time).
-
-  PRI1
-
 o a block device driver for ramdisks shared among virtual machines
 
   PRI3
_
